pkgs/applications/networking/mailreaders/notmuch/default.nix

```diff
@@ -25,15 +26,20 @@ stdenv.mkDerivation rec {
     sha256 = "0dfwa38vgnxk9cvvpza66szjgp8lir6iz6yy0cry9593lywh9xym";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [
+    pkgconfig
+    doxygen                   # (optional) api docs
+    pythonPackages.sphinx     # (optional) documentation -> doc/INSTALL
+    texinfo
+  ] ++ optional withEmacs [ emacs ];
+
   buildInputs = [
-    gnupg # undefined dependencies
+    gnupg                     # undefined dependencies
     xapian gmime talloc zlib  # dependencies described in INSTALL
-    doxygen perl  # (optional) api docs
-    pythonPackages.sphinx pythonPackages.python  # (optional) documentation -> doc/INSTALL
-    bash-completion  # (optional) dependency to install bash completion
-    emacs  # (optional) to byte compile emacs code, also needed for tests
-    ruby  # (optional) ruby bindings
+    bash-completion           # (optional) dependency to install bash completion
```

Comment:
Is  this invoked during build, or made a runtime reference? If the former is the case, it'd need to be in `nativeBuildInputs`.

Reply:
Hmm likely build only. Let me try

Reply:
Seems to work. I updated the PR

Reply:
Meh, I quickly tried by cross-compiling neomutt to aarch64, but this failed, as `talloc` required for `notmuch` currently fails to cross-compile.

I assume this is fine, we can always fix it up later.